lib/purescript-ast/src/Language/PureScript/Environment.hs

```diff
@@ -478,14 +450,14 @@ primSymbolTypes =
 primTypeErrorTypes :: M.Map (Qualified (ProperName 'TypeName)) (SourceType, TypeKind)
 primTypeErrorTypes =
   M.fromList $
-    [ (primSubName C.typeError "Doc", (kindType, ExternData))
-    , (primSubName C.typeError "Fail", (kindDoc -:> kindConstraint, ExternData))
-    , (primSubName C.typeError "Warn", (kindDoc -:> kindConstraint, ExternData))
-    , (primSubName C.typeError "Text", (kindSymbol -:> kindDoc, ExternData))
-    , (primSubName C.typeError "Quote", (kindType -:> kindDoc, ExternData))
-    , (primSubName C.typeError "QuoteLabel", (kindSymbol -:> kindDoc, ExternData))
-    , (primSubName C.typeError "Beside", (kindDoc -:> kindDoc -:> kindDoc, ExternData))
-    , (primSubName C.typeError "Above", (kindDoc -:> kindDoc -:> kindDoc, ExternData))
+    [ (primSubName C.typeError "Doc", (kindType, ExternData []))
+    , (primSubName C.typeError "Fail", (kindDoc -:> kindConstraint, ExternData [Phantom]))
+    , (primSubName C.typeError "Warn", (kindDoc -:> kindConstraint, ExternData [Phantom]))
```

Comment:
Those prim type classes have a phantom role but nominal roles are picked otherwise for type classes parameters here: https://github.com/purescript/purescript/blob/38dbf65db6c2720449571baaa654cedaae4cc2d6/src/Language/PureScript/TypeChecker.hs#L157

Reply:
Good spot, thanks! I agree these should probably be `Nominal`.